### DISCUSSION TITLE
Is there a way to increase message priority in JetStream ?

### DISCUSSION THREAD

**Discussion by AhmedAbouelkher** [nats-io/nats-server#3628] [Q&A]
Hi,
I've already pushed messages and not acked yet, so I need to prioritize one.

Is is possible?

**Comment by Jarema** [comment]
Hi @AhmedAbouelkher !

You mean you pushed:
```
Message 1
Message 2 
Message 3
```
and want to "prioritize" `Message 2`, so every consumer will get first `Message 2` before anything else?
Is that what you mean?

**Reply by AhmedAbouelkher** [reply]
Exactly

**Comment by Jarema** [comment]
You cannot alter the stream beyond deleting or purging (if that option is enabled in stream config), so the simple answer is no.

But you can do a lot of other things. 
For example, If you know the priority before publishing, you can have a subject token for high-priority messages, and a consumer dedicated to handling them, even in the same stream.

What is the use case?
Maybe that would allow us to give better answer.

**Reply by AhmedAbouelkher** [reply]
I'm currently operating a video processing microservice which had for the last 2 days more than 200 HD videos to process, one of our clients needs to publish his video ASAP and the video still waiting for his turn which will be in hours from now.

**Comment by Jarema** [comment]
If it is an emergency situation, you can always use API for getting specific message from the stream, process it and delete it / ignore it  afterwards.

For more permanent solution, I would probably create a separate subject for priority purposes.
Or load balance the processing across more microservices with Pull Consumer or Queue Push Consumer to not get into such delays in the future.

**Reply by yashsoni2** [reply]
Hi @Jarema 

I have a use case, where I want to priorities messages,

Suppose I pushed 1k messages in nats stream and want to send some more messages in same stream but those messages should process before them

is it achievable?? earlier i am using rabbitmq and internally it handle this and reorder the messages by setting priority in message header
